modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -292,9 +292,11 @@ public void refresh() {
                 return;
             }
             // Make sure that a readable symlink to the token file exists in the plugin config directory
+            // If it doesn't, the user doesn't wan to use AWS web identity tokens
```

Comment:
Can we change both comment lines to something like:
```suggestion
            // AWS_WEB_IDENTITY_ENV_VAR exists but we only use  Web Identity Tokens if a corresponding symlink exists and is readable
```